src/inc/til/some.h

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include <array>
+
+namespace til // Terminal Implementation Library. Also: "Today I Learned"
+{
+    template<class _Ty, size_t _Size>
+    class some : std::array<_Ty, _Size>
+    {
+    public:
+        using value_type = _Ty;
+        using size_type = size_t;
+        using difference_type = ptrdiff_t;
+        using pointer = _Ty*;
+        using const_pointer = const _Ty*;
+        using reference = _Ty&;
+        using const_reference = const _Ty&;
+
+        using iterator = _STD _Array_iterator<_Ty, _Size>;
+        using const_iterator = _STD _Array_const_iterator<_Ty, _Size>;
+
+        using reverse_iterator = _STD reverse_iterator<iterator>;
+        using const_reverse_iterator = _STD reverse_iterator<const_iterator>;
+
+        void fill(const _Ty& _Value)
+        {
+            array::fill(_Value);
+        }
+
+        void swap(some& _Other) noexcept(_Is_nothrow_swappable<_Ty>::value)
+        {
+            array::swap(_Other);
+            std::swap(_Used, _Other._Used);
+        }
+
+        _NODISCARD _CONSTEXPR17 iterator begin() noexcept

```

Comment:
we're always 17, so slap a constepxr on it all the time. we can't use `_NODISCARD`, as it's not a public API surface

Reply:
OK.